src/utils.ts

```diff
@@ -97,9 +99,14 @@ export const recursivelyDownloadTx = async (blockId: string, txIds: string[] = [
     return recursivelyDownloadTx(blockId, txIds, data);
   }
 
-  const newParents = parsedTx.parents.filter((parent) => txIds.indexOf(parent) < 0 && parent !== txId);
+  // check if we have already downloaded the parents
```

Comment:
Can you add another comment here explaining each condition? It was a long time since I've seen this code and I was trying to understand why did you add `!data.has(parent)`.

For what I understand you check `txIds` for the txs that are in the array to search but were not downloaded yet, `parent` if the tx is a parent of itself and added the `!data.has(parent)` in order to prevent downloading a tx that was already downloaded and is in the data map, right?

Reply:
Added the comments on 1ac06f4b1532df1cb3c1ac3cbca100cd91390bde

Exactly as you described. The data is serving as a "seen" list for the BFS, so we don't add transactions that we already downloaded to the queue